client/components/EntryFormRow/index.js

```diff
@@ -1,20 +1,57 @@
 import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { getCategories } from 'modules/categories';
+import { actions as AppActions } from 'modules/transactions';
 import DataSelector from './DataSelector';
 import styles from './style.scss';
 
+@connect(
+  state => ({
+    categories: getCategories(state)
+  }),
+  (dispatch => ({
+    actions: bindActionCreators(AppActions, dispatch)
+  }))
+)
 class EntryFormRow extends Component {
   static propTypes = {
-    categories: PropTypes.object.isRequired
+    categories: PropTypes.object.isRequired,
+    actions: PropTypes.object.isRequired
   }
 
   state = {
     categoryId: '16',
     description: '',
-    amount: ''
+    value: ''
   }
 
   handleFieldChange = e => this.setState({ [e.target.name]: e.target.value })
 
+  handleValueRefUpdate = (ref) => {
+    this.valueRef = ref;
+  }
+
+  addEntry = (e) => {
```

Comment:
I liked having it as a single handler since the only difference between the two was checking a key code. I'll break it up into two separate handlers for additional clarity sake.